### PR TITLE
Feature/bot 26 assing role based on the rank

### PR DESCRIPTION
**Pull request by fulviodenza**
This pr implements the set of the custom rank role creation.
This command creates a new role and can be performed just by an administrator.

### CODE REVIEW

**Comment by fulviodenza** [comment]
Files that need to be reviewed:

1. internal/commands/set_rank_role.go
2. internal/services/client.go
3. internal/services/user.go

**Review by Ladvace — APPROVED** [review]

File: internal/entities/server.go [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+	MaxLevel uint   `bson:"rankmaxlevel"`
 }
 
 type ServerType struct {
 	ServerID            string     `bson:"serverid"`
 	ServerPrefix        string     `bson:"guildprefix"`
 	NotificationChannel string     `bson:"notificationchannel"`
 	WelcomeChannel      string     `bson:"welcomechannel"`
-	CustomRanks         []RankType `bson:"customranks"`
+	CustomRanks         []RoleType `bson:"customranks"`
```
Comment (blarfoon) [inline internal/entities/server.go]: I think that struct name should match bson naming

File: internal/errors/errors.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package errors
+
+import "fmt"
+
+type DiscordError struct {
+	Message string
+}
+
+func (de DiscordError) Error() string {
```
Comment (blarfoon) [inline internal/errors/errors.go:7]: should this be `ToError` or `ToString` instead?

File: internal/services/user.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		v -= levelExp - user.CurrentLevelExperience
+		user.CurrentLevelExperience = 0
+		user.CurrentLevel += 1
+
+		customRank := db.GetCustomRanksByGuildId(u.ServerID)
+		sort.Slice(customRank, func(i, j int) bool {
+			return customRank[i].MinLevel > customRank[j].MinLevel
+		})
+
+		if len(customRank) > 0 {
```
Comment (blarfoon) [inline internal/services/user.go]: I feel like this check is kind of unnecessary

File: internal/services/user.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			for i, v := range customRank {
+				if user.CurrentLevel >= v.MinLevel && user.CurrentLevel < v.MaxLevel {
```
Comment (blarfoon) [inline internal/services/user.go]: overall I think these nested ifs can be rewritten in a clearer fashion

**Review by blarfoon — CHANGES_REQUESTED** [review]

**Review by blarfoon — APPROVED** [review]

**Pull request merged**